cpp/meta_module/algorithm/meta.cpp

```diff
@@ -0,0 +1,330 @@
+    return static_cast<int64_t>(data.relationship_types_cnt.size());
+  });
+  record.Insert(std::string(kReturnStats2).c_str(), relationship_type_count);
+  stats.Insert("relationshipTypeCount", mgp::Value(relationship_type_count));
+
+  int64_t property_key_count = LockAndExecuteReturn(metadata_mutex.property_key_cnt_mutex, online, [&]() {
+    return static_cast<int64_t>(data.property_key_cnt.size());
+  });
+  record.Insert(std::string(kReturnStats3).c_str(), property_key_count);
+  stats.Insert("propertyKeyCount", mgp::Value(property_key_count));
+
+  record.Insert(std::string(kReturnStats4).c_str(), data.node_cnt);
+  stats.Insert("nodeCount", mgp::Value(data.node_cnt));
+
+  record.Insert(std::string(kReturnStats5).c_str(), data.relationship_cnt);
+  stats.Insert("relationshipCount", mgp::Value(data.relationship_cnt));
+
+  auto CreateMap = [](const auto &map) {
+    mgp::Map result;
+    for (const auto &[key, value] : map) {
+      result.Insert(key, mgp::Value(value));
+    }
+    return result;
+  };
+
+  auto labels_map = LockAndExecuteReturn(metadata_mutex.labels_mutex, online, [&]() { return CreateMap(data.labels); });
+  record.Insert(std::string(kReturnStats6).c_str(), labels_map);
+  stats.Insert("labels", mgp::Value(std::move(labels_map)));
+
+  auto relationship_types_map = LockAndExecuteReturn(metadata_mutex.relationship_types_mutex, online,
+                                                     [&]() { return CreateMap(data.relationship_types); });
+  record.Insert(std::string(kReturnStats7).c_str(), relationship_types_map);
+  stats.Insert("relationshipTypes", mgp::Value(std::move(relationship_types_map)));
+
+  auto relationship_types_count_map = LockAndExecuteReturn(metadata_mutex.relationship_types_cnt_mutex, online,
+                                                           [&]() { return CreateMap(data.relationship_types_cnt); });
+  record.Insert(std::string(kReturnStats8).c_str(), relationship_types_count_map);
+  stats.Insert("relationshipTypesCount", mgp::Value(std::move(relationship_types_count_map)));
+
+  record.Insert(std::string(kReturnStats9).c_str(), stats);
+}
+
+}  // namespace
+
+void Update(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto created_objects{arguments[0].ValueList()};
+    const auto deleted_objects{arguments[1].ValueList()};
+    const auto removed_vertex_properties{arguments[2].ValueList()};
+    const auto removed_edge_properties{arguments[3].ValueList()};
+    const auto set_vertex_labels{arguments[4].ValueList()};
+    const auto removed_vertex_labels{arguments[5].ValueList()};
+
+    auto UpdateObject = [](const auto &objects, std::string_view vertex_event, std::string_view edge_event, int add) {
```

Comment:
Make this vertex_event and edge_event hardcoded as constant, there is no need to pass it as function parameter as it is not changing anywhere

Reply:
It changes depending whether it is called on created_objects or deleted_objects